### PR TITLE
dnn tensorflow support mirrorpad

### PR DESCRIPTION
**Pull request by vonchenplus**
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/622

dnn tensorflow support mirrorpad

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2019r1:16.04
build_image:Custom Win=openvino-2019r1
build_image:Custom Mac=openvino-2019r1

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
test_opencl:Custom=ON
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by alalek** [comment]
New features requires tests.
You can refer on adding tests for similar cases: #14315 (+testdata https://github.com/opencv/opencv_extra/pull/595)

File: modules/dnn/test/test_tf_importer.cpp [hunk 1]
```diff
@@ -146,6 +146,7 @@ TEST_P(Test_TensorFlow_layers, padding)
     runTensorFlowNet("padding_valid");
     runTensorFlowNet("spatial_padding");
     runTensorFlowNet("keras_pad_concat");
+    runTensorFlowNet("mirror_pad");
```
Comment (alalek) [inline modules/dnn/test/test_tf_importer.cpp]: Please open PR into [opencv_extra](https://github.com/opencv/opencv_extra) with test data (use the same source branch name as this PR: `ocv_mirrorpad`)

File: modules/dnn/src/tensorflow/tf_importer.cpp [hunk 2]
```diff
@@ -992,6 +993,35 @@ void TFImporter::populateNet(Net dstNet)
             if (getDataLayout(name, data_layouts) == DATA_LAYOUT_UNKNOWN)
                 data_layouts[name] = DATA_LAYOUT_NHWC;
         }
+        else if (type == "MirrorPad")
```
Comment (dkurt) [inline modules/dnn/src/tensorflow/tf_importer.cpp]: Please try to reduce code duplication. Try to integrate this padding mode into existing zero padding.

File: modules/dnn/test/test_tf_importer.cpp (hunk 1 above)
Comment (vonchenplus) [reply modules/dnn/test/test_tf_importer.cpp]: add pull request([622](https://github.com/opencv/opencv_extra/pull/622)) to opencv_extra

Comment (dkurt) [inline modules/dnn/test/test_tf_importer.cpp]: The prefix should be similar to testing files: here it's `mirror_pad` but in https://github.com/opencv/opencv_extra/pull/622/files it's `mirrorpad`.

Comment (vonchenplus) [reply modules/dnn/test/test_tf_importer.cpp]: Sorry, my mistake.

Comment (alalek) [reply modules/dnn/test/test_tf_importer.cpp]: @vonchenplus Please run test on your side with `--test_require_data=1 --gtest_filter=Test_TensorFlow_layers.padding/0` parameters

**Pull request merged**